### PR TITLE
[PDI-18834] Hard coded access and secret keys in a S3 URI no longer w…

### PR DESCRIPTION
**Pull request by ssamora**
…ork in 8.3

@pentaho-lmartins  @ddiroma @jgminder @peterrinehart

### CODE REVIEW

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 7m 13s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl legacy-amazon,s3-vfs
```
---
:ok_hand: All tests passed!
 
Tests run: **613**, Failures: **0**, Skipped: **0** &nbsp;&nbsp; [Test Results](http://wingman.pentaho.net/job/projects/job/big-data-plugin/job/PR-2129/1/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 6m 50s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl legacy-amazon,s3-vfs
```
---
:ok_hand: All tests passed!
 
Tests run: **613**, Failures: **0**, Skipped: **0** &nbsp;&nbsp; [Test Results](http://wingman.pentaho.net/job/projects/job/big-data-plugin/job/PR-2129/2/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 6m 23s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl legacy-amazon,s3-vfs
```
---
:ok_hand: All tests passed!
 
Tests run: **614**, Failures: **0**, Skipped: **0** &nbsp;&nbsp; [Test Results](http://wingman.pentaho.net/job/projects/job/big-data-plugin/job/PR-2129/3/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 2h 43m 4s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl legacy-amazon,s3-vfs
```
---
:ok_hand: All tests passed!
 
Tests run: **614**, Failures: **0**, Skipped: **0** &nbsp;&nbsp; [Test Results](http://wingman.pentaho.net/job/projects/job/big-data-plugin/job/PR-2129/4/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 5m 33s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl legacy-amazon,s3-vfs
```
---
:ok_hand: All tests passed!
 
Tests run: **614**, Failures: **0**, Skipped: **0** &nbsp;&nbsp; [Test Results](http://wingman.pentaho.net/job/projects/job/big-data-plugin/job/PR-2129/5/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

File: s3-vfs/src/main/java/org/pentaho/amazon/s3/S3Util.java [hunk 1]
```diff
@@ -42,6 +45,21 @@
...
+  private static final String URI_AWS_CREDENTIALS_REGEX = "(s3[an]?:\\/)?\\/(?<fullkeys>(?<keys>.*:.*)@)?s3[an]?\\/?(?<bucket>[^\\/]+)(?<path>.*)";
+
+  /** to be used with getKeysFromURI to get the FULL KEYS GROUP **/
+  private static final String URI_AWS_FULL_KEYS_GROUP = "fullkeys";
+
+  /** to be used with getKeysFromURI to get the KEYS GROUP **/
+  private static final String URI_AWS_KEYS_GROUP = "keys";
+
+  /** to be used with getKeysFromURI to get the BUCKET GROUP **/
+  private static final String URI_AWS_BUCKET_GROUP = "bucket";
```
Comment (bcostahitachivantara) [inline s3-vfs/src/main/java/org/pentaho/amazon/s3/S3Util.java:58]: Even with sonar issue makes sense keep it because it belongs to same context of regex.

**Review by bcostahitachivantara — APPROVED** [review]

**Pull request merged**